examples/example-gauth/src/main/java/io/grpc/examples/googleAuth/GoogleAuthClient.java

```diff
@@ -0,0 +1,137 @@
+ * limitations under the License.
+ */
+
+package io.grpc.examples.googleAuth;
+
+import com.google.auth.oauth2.AccessToken;
+import com.google.auth.oauth2.GoogleCredentials;
+import io.grpc.CallCredentials;
+import io.grpc.auth.MoreCallCredentials;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.StatusRuntimeException;
+import com.google.pubsub.v1.PublisherGrpc;
+import com.google.pubsub.v1.ListTopicsResponse;
+import com.google.pubsub.v1.ListTopicsRequest;
+
+import java.io.FileInputStream;
+import java.util.Arrays;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Example to illustrate use of Google credentials as described in
+ * @see <a href="../../../../../../GOOGLE_AUTH_EXAMPLE.md">Google Auth Example README</a>
+ *
+ * Also @see <a href="https://cloud.google.com/pubsub/docs/reference/rpc/">Google Cloud Pubsub via gRPC</a>
+ *
+ */
+public class GoogleAuthClient {
+  private static final Logger logger = Logger.getLogger(GoogleAuthClient.class.getName());
+
+  private final ManagedChannel channel;
+
+  /**
+   * stub generated from the proto file.
+   */
+  private final PublisherGrpc.PublisherBlockingStub blockingStub;
+
+  /**
+   * Construct our gRPC client that connects to the pubsub server at {@code host:port}.
+   *
+   * @param host  host to connect to - typically "pubsub.googleapis.com"
+   * @param port  port to connect to - typically 443 - the TLS port
+   * @param callCredentials  the Google call credentials created from a JSON file
+   */
+  public GoogleAuthClient(String host, int port, CallCredentials callCredentials) {
+    // Google API invocation requires a secure channel. Channels are secure by default (SSL/TLS)
+    this(ManagedChannelBuilder.forAddress(host, port).build(), callCredentials);
+  }
+
+  /**
+   * Construct our gRPC client that connects to the pubsub server using an existing channel.
+   *
+   * @param channel    channel that has been built already
+   * @param callCredentials  the Google call credentials created from a JSON file
+   */
+  GoogleAuthClient(ManagedChannel channel, CallCredentials callCredentials) {
+    this.channel = channel;
+    blockingStub = PublisherGrpc.newBlockingStub(channel).withCallCredentials(callCredentials);
+  }
+
+  public void shutdown() throws InterruptedException {
+    channel.shutdown().awaitTermination(5, TimeUnit.SECONDS);
+  }
+
+  /**
+   * Get topics (max 10) for our project ID: the topic list is logged to the logger.
+   *
+   * @param projectID  the GCP project ID to get the pubsub topics for. This is a string like
+   *                   "projects/balmy-cirrus-225307" where "balmy-cirrus-225307" is
+   *                   the project ID for the project you created.
+   */
+  public void getTopics(String projectID) {
+    logger.info("Will try to get topics for poroject " + projectID + " ...");
```

Comment:
nit: use `logger.log(Level.INFO, "{}", )` pattern to avoid string concat if logging level is higher than INFO.